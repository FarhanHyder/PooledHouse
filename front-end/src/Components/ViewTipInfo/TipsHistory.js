import React from 'react';

const TipsHistory = (props) => {
    const tips = props.tipsInfo;
    let view =tips.map(tips => {
        return (
            <div>
                <div>{tips.business_street_address}</div>
                <div>{tips.shift_position}</div>
                <div>{tips.shift_time}</div>
                <div>${Number.parseFloat(tips.takehome/tips.shift_length).toFixed(2)}</div>
            </div>
        );
    });

    return (
        <div>
            <div>
                <div>Locations</div>
                <div>Position</div>
                <div>Shift</div>
                <div>Tips</div>
            </div>
            {view}
        </div>
    );
}

export default TipsHistory;