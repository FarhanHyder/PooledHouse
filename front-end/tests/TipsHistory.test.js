import React from 'react';
import renderer from 'react-test-renderer';
import TipsHistory from '../src/Components/ViewTipInfo/ViewTipsHistory';

test('TipsHistory should render correct for single element', () => {
  const thisProps = {
    Poolinos: [
      {
        address: '292 Bowery, New York, NY 10012',
        position: 'Busser',
        tips: 87,
        hour: 5,
        shift: 'PM',
        neighborhood: 'Soho',
        date: '04/30/2019'
      }
    ]
  };

  const component = renderer.create(
    <TipsHistory tipsHistory={thisProps.Poolinos} />
  );

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});


test('TipsHistory should render correct for multiple elements', () => {
  const thisProps = {
    Poolinos: [
      {
        address: '292 Bowery, New York, NY 10012',
        position: 'Busser',
        tips: 87,
        hour: 5,
        shift: 'PM',
        neighborhood: 'Soho',
        date: '04/30/2019'
      },
      {
        address: '292 Bowery, New York, NY 10012',
        position: 'Busser',
        tips: 93,
        hour: 6,
        shift: 'PM',
        neighborhood: 'Soho',
        date: '05/02/2019'
      }
    ]
  };

  const component = renderer.create(
    <TipsHistory tipsHistory={thisProps.Poolinos} />
  );

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});