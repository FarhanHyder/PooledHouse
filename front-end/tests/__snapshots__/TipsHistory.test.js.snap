// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TipsHistory should render correct for multiple elements 1`] = `
<div
  className="card container border border-primary rounded-lg"
>
  <div
    className="table-responsive-sm"
  >
    <table
      className="table table-dark table-striped table-bordered table-hover"
    >
      <thead
        className="bg-secondary"
      >
        <tr>
          <th>
            #
          </th>
          <th>
            Locations
          </th>
          <th>
            Position
          </th>
          <th>
            Shift
          </th>
          <th>
            Hours Worked
          </th>
          <th>
            Tips
          </th>
        </tr>
      </thead>
      <tbody>
        <tr>
          <td>
            1
          </td>
          <td>
            292 Bowery, New York, NY 10012
          </td>
          <td>
            Busser
          </td>
          <td>
            PM
          </td>
          <td>
            5
             Hours
          </td>
          <td>
            $
            87
          </td>
        </tr>
      </tbody>
      <tbody>
        <tr>
          <td>
            2
          </td>
          <td>
            292 Bowery, New York, NY 10012
          </td>
          <td>
            Busser
          </td>
          <td>
            PM
          </td>
          <td>
            6
             Hours
          </td>
          <td>
            $
            93
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;

exports[`TipsHistory should render correct for single element 1`] = `
<div
  className="card container border border-primary rounded-lg"
>
  <div
    className="table-responsive-sm"
  >
    <table
      className="table table-dark table-striped table-bordered table-hover"
    >
      <thead
        className="bg-secondary"
      >
        <tr>
          <th>
            #
          </th>
          <th>
            Locations
          </th>
          <th>
            Position
          </th>
          <th>
            Shift
          </th>
          <th>
            Hours Worked
          </th>
          <th>
            Tips
          </th>
        </tr>
      </thead>
      <tbody>
        <tr>
          <td>
            1
          </td>
          <td>
            292 Bowery, New York, NY 10012
          </td>
          <td>
            Busser
          </td>
          <td>
            PM
          </td>
          <td>
            5
             Hours
          </td>
          <td>
            $
            87
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;
